server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/RabbitMQMailQueueTest.java

```diff
@@ -196,10 +196,10 @@ void browseAndDequeueShouldCombineWellWhenDifferentSlices() throws Exception {
 
         @Test
         void mailQueueShouldBeInitializedWhenCreating(CassandraCluster cassandra) {
-            String name = "myQueue";
+            org.apache.james.queue.api.MailQueueName name = org.apache.james.queue.api.MailQueueName.of("myQueue");
```

Comment:
Why is rabbitMQ using a different MQ name?

Can't the 2 POJOs be reworked into one?

Reply:
![Screenshot from 2020-02-21 08-58-33](https://user-images.githubusercontent.com/708318/75015223-01be0980-5480-11ea-999c-3d907c02c118.png)

I chose not to refactor this because rabbitmq MailQueueName has specific features.

If you think I should merge both concepts, please tell me how I'm supposed to expose those features without degrading code cohesion.

Reply:
I was thinking about moving the specific methods as static methods in WorkQueueName or ExchangeName and hopefully being able to then merge the two into one.

That is something I would be trying, not sure what the end result is, but I find having 2 classes representing the same concept pretty misleading.

Reply:
WorkQueueName and ExchangeName share the way to prefix queue name. 

They also have in common the fact they only make sense for RabbitMQ.

Removing the MailQueueName for rabbitmq has the consequence of splitting the two subclasses (and then, how do you share stuff?). 

I really think we would loose cohesion.